packages/3dom/src/protocol.ts

```diff
@@ -0,0 +1,88 @@
+/* @license
+ * Copyright 2020 Google LLC. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the 'License');
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an 'AS IS' BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {RGBA} from './api.js';
+
+/**
+ * The protocol between 3DOM execution contexts is strictly defined.
+ * Only specific types of messages are allowed, and their types are
+ * all included in the ThreeDOMMessageType map.
+ */
+export const ThreeDOMMessageType = {
+  // Used when the host execution context and scene graph execution context
+  // are negotiating a connection
+  HANDSHAKE: 1,
+
+  // Notification sent to the host execution context to indicate that the
+  // scene graph execution context has finished initializing
+  CONTEXT_INITIALIZED: 2,
+
+  // A message that indicates that a custom script is meant to be imported
+  // into the scene graph execution context
+  IMPORT_SCRIPT: 3,
+
+  // A notification from the host execution context that the main Model has
+  // changed, including the sparse, serialized scene graph of the new Model
+  MODEL_CHANGED: 4,
```

Comment:
It's when the global model changes. Increasingly, I think `GLOBAL_MODEL_CHANGED` better, given that we are already talking about loading "sub-models" into the scene.

Reply:
Kind of like my comment on changeModel above, this sounds incremental to me. Maybe `NEW_GLOBAL_MODEL`? Also, this serialized scene graph will not be sparse, right, since it's the whole model?

Reply:
Sparse may be overused, but I intend it to mean that only a skeleton representation of the graph is included (we know most of the properties to fill out every object comprehensively, but we only send a subset).